compiler/rustc_mir_build/src/build/expr/as_place.rs

```diff
@@ -299,6 +299,17 @@ impl<'tcx> PlaceBuilder<'tcx> {
         to_upvars_resolved_place_builder(self, tcx, typeck_results).unwrap()
     }
 
+    fn try_upvars_resolved<'a>(
+        self,
+        tcx: TyCtxt<'tcx>,
+        typeck_results: &'a ty::TypeckResults<'tcx>,
+    ) -> PlaceBuilder<'tcx> {
```

Comment:
This should return a result too. We want the callers of this function to decide what to do in case its an error.